common/remote/hooks/azure/azurefile.cpp

```diff
@@ -459,12 +459,15 @@ AzureFile::AzureFile(const char *_azureFileName) : fullName(_azureFileName)
 
         StringBuffer planeName(slash-filename, filename);
         Owned<IPropertyTree> plane = getStoragePlane(planeName);
+        if (!plane)
+            throw makeStringExceptionV(99, "Unknown storage plane %s", planeName.str());
+
         const char * api = plane->queryProp("storageapi/@type");
         if (!api)
             throw makeStringExceptionV(99, "No storage api defined for plane %s", planeName.str());
 
         constexpr size_t lenPrefix = strlen(azureBlobPrefix);
-        if ((strncmp(api, azureBlobPrefix, lenPrefix-1) != 0) || api[lenPrefix-1] != ':')
+        if ((strncmp(api, azureBlobPrefix, lenPrefix-1) != 0))
```

Comment:
I think the extra test should have been
```
api[lenPrefix-1] == 0
```
so that it doesn't match azureblobv2

possibly better would be to define a string without the trailing colon and use strieq.

Reply:
Changed to use strieq.